// dbweb/core/module-webpack/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var path_1 = __importDefault(require("path"));
function config(dirname, packageName) {
    var plugins = [];
    for (var _i = 2; _i < arguments.length; _i++) {
        plugins[_i - 2] = arguments[_i];
    }
    return {
        entry: path_1.default.join(__dirname, "src/index.js"),
        output: {
            filename: 'index.js',
            libraryTarget: "commonjs2"
        },
        module: {
            rules: [
                {
                    test: /\.(js|jsx)$/,
                    use: "babel-loader",
                    exclude: /(node_modules|dist)/
                },
                {
                    test: /\.css$/,
                    use: ["style-loader", "css-loader"]
                },
                {
                    test: /\.(png|jpg|gif)$/,
                    use: [
                        {
                            loader: 'url-loader',
                            options: {
                                limit: 8192,
                                outputPath: path_1.default.join("static", packageName, "images"),
                                publicPath: "/front/static/" + packageName + "/images"
                            }
                        }
                    ]
                }
            ]
        },
        plugins: plugins,
        // plugins: [htmlWebpackPlugin],
        resolve: {
            extensions: [".js", ".jsx"]
        },
        devServer: {
            port: 3001
        },
        externals: {
            'react': 'commonjs react',
            'material-ui': 'commonjs material-ui',
            "react-dom": 'commonjs react-dom'
        }
    };
}
exports.config = config;
;
