// dbweb/modules/common/front/webpack.config.js
const path = require('path');
var PACKAGE = require('./package.json');
// const HtmlWebpackPlugin = require("html-webpack-plugin");

// const htmlWebpackPlugin = new HtmlWebpackPlugin({
// 	template: path.join(__dirname, "examples/src/index.html"),
// 	filename: "./index.html"
// });
const CleanWebpackPlugin = require('clean-webpack-plugin');
module.exports = {
	entry: path.join(__dirname, "src/index.js"),
	output: {
		filename: 'index.js',
		libraryTarget: "commonjs2"
	},
	module: {
		rules: [
			{
				test: /\.(js|jsx)$/,
				use: "babel-loader",
				exclude: /(node_modules|dist)/
			},
			{
				test: /\.css$/,
				use: ["style-loader", "css-loader"]
			},
			{
				test: /\.(png|jpg|gif)$/,
				use: [
					{
						loader: 'url-loader',
						options: {
							limit: 8192,
							outputPath:path.join("static",PACKAGE.name,"images"),
							publicPath:"/front/static/"+PACKAGE.name+"/images"
						}
					}
				]
			}
		]
	},
	plugins: [new CleanWebpackPlugin(['dist'])],
	// plugins: [htmlWebpackPlugin],
	resolve: {
		extensions: [".js", ".jsx"]
	},
	devServer: {
		port: 3001
	},
	externals: {
		'react': 'commonjs react',
		'material-ui': 'commonjs material-ui',// this line is just to use the React dependency of our parent-testing-project instead of using our own React.
		"react-dom": 'commonjs react-dom'
	}
};
