// dbweb/modules/common/front/src/login.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';
const styles = theme => ({
	root: theme.mixins.gutters({
		paddingTop: 16,
		paddingBottom: 16,
		marginTop: theme.spacing.unit * 3,
	}),
	textField: {
		marginLeft: theme.spacing.unit,
		marginRight: theme.spacing.unit,
		width: 200,
	},
});

class Login extends Component {
	render() {
		const {classes}=this.props;
		return (
			<Paper className={classes.root}>
				<Typography variant="headline" component="h3">
					请输入用户名密码登录
				</Typography>
				<TextField
					id="password-input"
					label="Password"
					className={classes.textField}
					type="password"
					autoComplete="current-password"
					margin="normal"
				/>
			</Paper>
		);
	}
}
Login.propTypes = {
	classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Login);
